src/ppx/geometry.cpp

```diff
@@ -13,12 +13,396 @@
 // limitations under the License.
 
 #include "ppx/geometry.h"
+#include <cmath>
 
 #define NOT_INTERLEAVED_MSG "cannot append interleaved data if attribute layout is not interleaved"
 #define NOT_PLANAR_MSG      "cannot append planar data if attribute layout is not planar"
 
 namespace ppx {
 
+// -------------------------------------------------------------------------------------------------
+// VertexDataProcessorBase
+//     interface for all VertexDataProcessors
+//     with helper functions to allow derived classes to access Geometry
+// Note that the base and derived VertexDataProcessor classes do not have any data member
+//     be careful when adding data members to any of these classes
+//     it could create problems for multithreaded cases for multiple geometry objects
+// -------------------------------------------------------------------------------------------------
+class VertexDataProcessorBase
+{
+public:
+    // validates the layout
+    // returns false if the validation fails
+    virtual bool Validate(Geometry* pGeom) = 0;
+    // update the vertex buffer and the vertex buffer index
```

Comment:
Updates?
Assuming the function comment was followed by "This function..." 
and same for the next function